### PR TITLE
fix flare rc switch action with flight option bit 10 active

### PR DESCRIPTION
**Pull request by Hwurzburg**
found this while setting up a new tri-tiltrotor....unintended consequence of FLIGHT_OPTION bit 10 in conjunction with the FLARE rc option( RC trim in no longer zero thrust)...tested on TVBS and Convergence in RF SITL with and without the flight option enabled for flare switch

also fixes the QUADPLANE build option boundaries in this function

### CODE REVIEW

**Comment by IamPete1** [comment]
This same issue is likely to come up in other places. Looks like here at least:

https://github.com/ArduPilot/ardupilot/blob/86c481b4b094d136ad0224ec6344a55d96ee333d/ArduPlane/Attitude.cpp#L195

I suggest we fix `get_adjusted_throttle_input` to allow deadzone in the mid throttle case, currently it does not work. 

https://github.com/ArduPilot/ardupilot/blob/86c481b4b094d136ad0224ec6344a55d96ee333d/ArduPlane/reverse_thrust.cpp#L149

to something like:
```
float thr_in = no_deadzone ? channel_throttle->norm_input() : channel_throttle->norm_input_dz();
float ret = channel_throttle->get_range() * throttle_curve(aparm.throttle_cruise * 0.01, 0, 0.5 + 0.5*thr_in);
```

The we can change both the flare and the case I found in `Attitude.cpp` to be: 

`is_zero(get_adjusted_throttle_input(false))`

As ever I have not tested where my proposed approach actually works...

**Comment by IamPete1** [comment]
In fact I think it my method probably won't, it still applies the deadzone about trim. You can be sure the throttle is zero at least. You could alternately get the scaled servo output and see if that is zero I guess.

**Comment by Hwurzburg** [comment]
nice catch....pushed a fix for the attitude....had forgotten that it was a three position switch...the attitude option is for future expansion into a real automated flare that Tridge might undertake at some time...personally, I never use it...I want control of the pitch during the landings in which I use the tilts up ...which is often when flying my TVBSs

**Comment by Hwurzburg** [comment]
added a fix to allow checking of flare tilts on tilt rotors when disarmed....worked for tailsitters but not tilt rotors until armed and in air...

File: ArduPlane/Attitude.cpp [hunk 1]
```diff
@@ -190,11 +190,17 @@ float Plane::stabilize_pitch_get_pitch_out(float speed_scaler)
...
+       - throttle stick at zero thrust
+       - in fixed wing non auto-throttle mode
+    */
     if (!quadplane_in_transition &&
         !control_mode->is_vtol_mode() &&
-        channel_throttle->in_trim_dz() &&
         !control_mode->does_auto_throttle() &&
-        flare_mode == FlareMode::ENABLED_PITCH_TARGET) {
+        flare_mode == FlareMode::ENABLED_PITCH_TARGET &&
+        ((!(g2.flight_options & FlightOptions::CENTER_THROTTLE_TRIM) && channel_throttle->in_trim_dz()) ||
```
Comment (tridge) [inline ArduPlane/Attitude.cpp:202]: add bool function for throttle_is_zero() common in radio.cpp

File: ArduPlane/Attitude.cpp [hunk 2]
```diff
@@ -190,11 +190,17 @@ float Plane::stabilize_pitch_get_pitch_out(float speed_scaler)
     if (control_mode == &mode_stabilize && channel_pitch->get_control_in() != 0) {
         disable_integrator = true;
     }
+    /* force landing pitch if:
```
Comment (peterbarker) [inline ArduPlane/Attitude.cpp:193]: Maybe create a `should_force_landing_pitch` method?

**Review by tridge — CHANGES_REQUESTED** [review]

File: ArduPlane/Attitude.cpp (hunk 1 above)
Comment (Hwurzburg) [reply ArduPlane/Attitude.cpp]: is this what you wanted?

**Review by tridge — APPROVED** [review]

File: libraries/RC_Channel/RC_Channel.cpp [hunk 3]
```diff
@@ -326,6 +326,15 @@ bool RC_Channel::in_trim_dz() const
     return is_bounded_int32(radio_in, radio_trim - dead_zone, radio_trim + dead_zone);
 }
 
+
+/*
+   return trues if input is within deadzone of min
+*/
+bool RC_Channel::within_min_dz() const
+{
+    return is_bounded_int32(radio_in, radio_min - dead_zone, radio_min + dead_zone);
```
Comment (IamPete1) [inline libraries/RC_Channel/RC_Channel.cpp:335]: I wonder if this should just be:
```suggestion
    return radio_in < radio_min + dead_zone;
```
Seems a bit odd to apply the deadzone on the low side of min.

File: ArduPlane/servos.cpp [hunk 4]
```diff
@@ -764,23 +764,25 @@ void Plane::servos_twin_engine_mix(void)
...
+       -at zero thrust: in throttle trim dz except for sprung throttle option where trim is at hover stick
+    */
+    if (!control_mode->does_auto_throttle() && flare_mode != FlareMode::FLARE_DISABLED && throttle_at_zero()) {
+            int32_t tilt = -SERVO_MAX;  //this is tilts up for a normal tiltrotor if at zero thrust throttle stick      
+            if (quadplane.tiltrotor.enabled() && (quadplane.tiltrotor.type == Tiltrotor::TILT_TYPE_BICOPTER)) {
+                tilt = 0; // this is tilts up for a Bicopter
+            }
+            if (quadplane.tailsitter.enabled()) {
+                tilt = SERVO_MAX; //this is tilts up for a tailsitter
+            }
```
Comment (IamPete1) [inline ArduPlane/servos.cpp:785]:
```suggestion
        int32_t tilt = -SERVO_MAX;  //this is tilts up for a normal tiltrotor if at zero thrust throttle stick      
        if (quadplane.tiltrotor.enabled() && (quadplane.tiltrotor.type == Tiltrotor::TILT_TYPE_BICOPTER)) {
            tilt = 0; // this is tilts up for a Bicopter
        }
        if (quadplane.tailsitter.enabled()) {
            tilt = SERVO_MAX; //this is tilts up for a tailsitter
        }
```
Double indentation

**Review by IamPete1 — APPROVED** [review]
LGTM

**Comment by Hwurzburg** [comment]
> LGTM

merge on ci pass?

**Pull request merged**